src/meta/src/hummock/manager/commit_epoch.rs

```diff
@@ -161,16 +188,19 @@ impl HummockManager {
             self.env.notification_manager(),
             &self.metrics,
         );
+        let mut compaction_group_manager =
+            compaction_group_manager_guard.start_compaction_groups_txn();
```

Comment:
Can we only acquire the `compaction_group_manager` lock only when `batch_commit_for_new_cg` is not empty in L222?

Reply:
Resolved.

The original `BTreeMapTransaction` cannot fulfill this, because it holds the mutation reference to the RwLockGuard, and they cannot be created and stored together, and otherwise there will be the problem of self-referencing.

In this PR, I change to let `BTreeMapTransaction` support holding any pointer type that impl `DerefMut<BTreeMap>`, so that the ownership of `RwLockGuard` can be held by `BTreeMapTransaction` and there won't be self-referencing. We further introduce a new struct `DerefMutForward`, which forward `&mut CompactionGroupManager` to `&mut BTreeMap<...>`. This is required because `RwLockGuard` wraps `CompactionGroupManager`, but we actually need `impl DerefMut<BTreeMap<...>>`.